src/librustc_mir/hair/pattern/_match.rs

```diff
@@ -656,11 +1012,42 @@ pub fn is_useful<'p, 'a: 'p, 'tcx: 'a>(cx: &mut MatchCheckCtxt<'a, 'tcx>,
             pat_constructors(cx, row[0], pcx).unwrap_or(vec![])
         }).collect();
         debug!("used_ctors = {:#?}", used_ctors);
+        // `all_ctors` are all the constructors for the given type, which
+        // should all be represented (or caught with the wild pattern `_`).
         let all_ctors = all_constructors(cx, pcx);
         debug!("all_ctors = {:#?}", all_ctors);
-        let missing_ctors: Vec<Constructor> = all_ctors.iter().filter(|c| {
-            !used_ctors.contains(*c)
-        }).cloned().collect();
+
+        // `missing_ctors` are those that should have appeared
```

Comment:
Bad comment (given that `missing_ctors` can be covered by a wildcard).

Better:
```Rust
// `missing_ctors` are constructors that are not matched against by any non-wildcard
// pattern in our column.
```

Reply:
Oh, good point: I wrote this before reading the paper and forgot to come back to it.